sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVInferSchema.scala

```diff
@@ -108,14 +109,38 @@ private[csv] object CSVInferSchema {
   }
 
   private def tryParseDouble(field: String): DataType = {
-    if ((allCatch opt field.toDouble).isDefined) {
+    val doubleTry = allCatch opt field.toDouble
```

Comment:
Hm.. then wouldn't any data be inferred as `DoubleType`?
I mean, if `DecimalType` is checked first then any numeric types having the lower priority would not be inferred because all the numeric types can be represented by `DecimalType`.